posts/Gavin/2020-12-01-hibernate-reactive.adoc

```diff
@@ -0,0 +1,76 @@
+= "Introducing Hibernate Reactive"
+Gavin King
+:awestruct-tags: [ "Hibernate Reactive" ]
+:awestruct-layout: blog-post
+---
+
+:HR: https://github.com/hibernate/hibernate-reactive
+:vertx: https://vertx.io/docs/#data_access
+:Mutiny: https://smallrye.io/smallrye-mutiny/
+:docs: https://github.com/hibernate/hibernate-reactive/blob/master/documentation/src/main/asciidoc/reference/introduction.adoc
+
+Reactive programming lets us take advantage of non-blocking IO to reduce resource
+usage in our programs. For a long time, non-blocking IO was something that just
+wasn't an option for programs which interact with relational databases, since data
+access APIs like JDBC and ODBC were designed around a fundamentally synchronous,
+blocking paradigm.
+
+That's changed quite recently, with initiatives like the {vertx}[Vert.x client APIs]
+for PostgreSQL, MySQL, Db2, and SQL Server, along with other similar projects. Thus,
+it's now possible for a Java program to interact with a relational database in a
+completely asynchronous, non-blocking fashion.
+
+Furthermore, APIs like {Mutiny}[Mutiny] simplify reactive programming using streams
+instead of nested callbacks, alleviating at least some of the pain of asynchronous
+programming in Java.
+
+But of course back in 2006, when we designed the Java Persistence API to simplify
+ORM in Java and unify the ORM implementations that existed at the time, "reactive
+programming" wasn't on anyone's radar. So JPA offers an API that is fundamentally
+blocking in nature. "Reactive ORM" is something that simply hasn't existed.
+
+So today we're introducing the community to {HR}[Hibernate Reactive], a reactive
+API for Hibernate ORM, which supports non-blocking database clients and reactive
+programming as a paradigm for interacting with the relational database.
+
+Hibernate Reactive reuses most of the implementation of Hibernate ORM, but replaces
+the code which interacts with JDBC with a new layer designed around reactive streams,
+and exposes a new reactive `Session` API to the programmer.
+
+In fact, there's _two_ reactive `Session` APIs, one for programs using Mutiny, and
```

Comment:
Would love to see at least one simple example for how the API usage looks like.

Reply:
I agree, I was already thinking that. Check staging.